src/modules/endpoint/reducers/switchLanguage.js

```diff
@@ -1,96 +1,90 @@
 // @flow
 
-import type { CategoriesStateType, RouteStateType } from '../../app/StateType'
+import type { CategoryRouteStateType, CityContentStateType, StateType } from '../../app/StateType'
 import { defaultRouteState } from '../../app/StateType'
-import type { SwitchCategoryLanguageActionType } from '../../app/StoreActionType'
+import type { CityContentLoadedActionType } from '../../app/StoreActionType'
 import { mapValues } from 'lodash/object'
 import { reduce } from 'lodash/collection'
-import { CategoryModel } from '@integreat-app/integreat-api-client'
+import { CategoriesMapModel, CategoryModel } from '@integreat-app/integreat-api-client'
 
-const translatePath = (model: CategoryModel, currentCity: string, newLanguage: string) => {
+const translatePath = (model: CategoryModel, currentCity: string, newLanguage: string): string | null => {
   if (model.id === 0) {
     return `/${currentCity}/${newLanguage}`
   } else {
     return model.availableLanguages.get(newLanguage)
   }
 }
 
-const translateChildren = (models, newCategoriesMap, children, currentCity, newLanguage) =>
-  reduce(children, (result, value: Array<string>, path: string) => {
-    const translatedKey = translatePath(models[path], currentCity, newLanguage)
+const translateChildren = (
+  models: Array<CategoryModel>, newCategoriesMap: CategoriesMapModel,
+  children: Array<string>, city: string, newLanguage: string
+): { [path: string]: Array<string> } =>
+  reduce(children, (acc, value: Array<string>, path: string) => {
+    const newPath = translatePath(models[path], city, newLanguage)
 
-    if (!translatedKey) {
+    if (!newPath) {
```

Comment:
This code was valid imo and these changes are unnecessary, because the changes only address the style.

Reply:
Yep, I started refactoring it, but ran out of time...

Reply:
I meant that we should only refactor stuff that is necessary. Keeping the changes minimal keeps the git history clean ;)

Reply:
Sure, but I think this file will look completely different once switching language works again^^